### PR TITLE
Change Trajectory to properly deal with forks on empty timelines

### LINKED ISSUE #483: NewFork at nonexistent time
Found at commit 60e70b7 while exercising #463, at very high warp (1000x? 10000x?) when stopping/starting the plugin repeatedly.  This didn't show up at lower warp (100x).

```
Log line format: [IWEF]mmdd hh:mm:ss.uuuuuu threadid file:line] msg
    @   2831A327  	google::LogMessageFatal::~LogMessageFatal [0x2831A326+102] (c:\users\phl\projects\github\google\glog-0.3.3\src\logging.cc:1995)
    @   282E9847  	principia::physics::Trajectory<principia::geometry::Frame<enum principia::serialization::Frame_PluginTag,3,1> >::NewFork [0x282E9846+230] (c:\users\phl\projects\github\principia\physics\trajectory_body.hpp:165)
    @   282FDEFE  	principia::ksp_plugin::Plugin::UpdatePredictions [0x282FDEFD+349] (c:\users\phl\projects\github\principia\ksp_plugin\plugin.cpp:773)
    @   282F7300  	principia::ksp_plugin::Plugin::AdvanceTime [0x282F72FF+1007] (c:\users\phl\projects\github\principia\ksp_plugin\plugin.cpp:177)
    @   282D7A41  	principia__AdvanceTime [0x282D7A40+80] (c:\users\phl\projects\github\principia\ksp_plugin\interface.cpp:237)
F0323 22:46:42.413811  4460 trajectory_body.hpp:165] Check failed: fork_it != timeline_.end() NewFork at nonexistent time
```
[INFO file](https://gist.github.com/pleroy/4e127883fccd4b31497b#file-info)

### PR DESCRIPTION
**Pull request by pleroy**
Fix #483.

### CODE REVIEW

File: physics/trajectory_body.hpp [hunk 1]
```diff
@@ -146,7 +139,7 @@ template<typename Frame>
 void Trajectory<Frame>::ForgetBefore(Instant const& time) {
   // Check that this is a root.
   CHECK(is_root()) << "ForgetBefore on a nonroot trajectory";
-  // Check that |time| is the time of one of our Timeline or the time of fork.
+  // Check that |time| is the time of one of our Timeline.
```
Comment (eggrobin) [inline physics/trajectory_body.hpp]: swap

**Comment by eggrobin** [comment]
As discussed, it probably doesn't work, a fork B at the forkpoint of a fork A should behave the same independently of whether A is empty or not.

File: physics/trajectory_body.hpp (hunk 1 above)
Comment (pleroy) [inline physics/trajectory_body.hpp]: Added a comment on this topic.

**Comment by pleroy** [comment]
Fixed the management of iterators so that the problem mentioned in the above comment doesn't arise.  Added tests to check this.

**Pull request merged**